server/routes/restaurantRoutes.js

```diff
@@ -23,11 +23,29 @@ router.get('/test', async (req, res) => {
     querySnapshot.forEach(doc => {
       allRestaurants.push(doc.data());
     });
-    
+
     res.json(allRestaurants);
   } catch (error) {
     res.status(400).json({ message: 'Failed to retrieve restaurants.' });
   }
 });
 
+
+//only queries for 1 single restaurant
+router.get('/restaurant', async (req, res) => {
+  // console.log(req.query);
+  const q = query(restaurantsRef, where("restaurantName","==", req.query.restaurant));
```

Comment:
Anything that can fail should be in the `try` block. Even though this page should never be loaded directly, if you do, it crashes the server, which is not good.

Reply:
done